nucypher-core-wasm/src/lib.rs

```diff
@@ -542,6 +542,116 @@ impl EncryptedTreasureMap {
     }
 }
 
+//
+// Request Keys
+//
+
+#[wasm_bindgen]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct RequestSharedSecret(nucypher_core::RequestSharedSecret);
+
+#[wasm_bindgen]
+#[derive(PartialEq, Eq, Debug, derive_more::From, derive_more::AsRef)]
+pub struct RequestPublicKey(nucypher_core::RequestPublicKey);
+
+#[wasm_bindgen]
+impl RequestPublicKey {
+    #[wasm_bindgen(js_name = fromBytes)]
+    pub fn from_bytes(data: &[u8]) -> Result<RequestPublicKey, Error> {
+        from_bytes::<_, nucypher_core::RequestPublicKey>(data)
+    }
+
+    #[wasm_bindgen(js_name = toBytes)]
+    pub fn to_bytes(&self) -> Box<[u8]> {
+        to_bytes(self)
+    }
+
+    #[allow(clippy::inherent_to_string)]
+    #[wasm_bindgen(js_name = toString)]
+    pub fn to_string(&self) -> String {
+        format!("{}", self.0)
+    }
+
+    pub fn equals(&self, other: &RequestPublicKey) -> bool {
+        self.0 == other.0
+    }
+}
+
+#[wasm_bindgen]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct RequestSecretKey(nucypher_core::RequestSecretKey);
+
+#[wasm_bindgen]
+impl RequestSecretKey {
+    /// Generates a secret key using the default RNG and returns it.
+    pub fn random() -> Self {
+        Self(nucypher_core::RequestSecretKey::random())
+    }
+
+    /// Generates a secret key using the default RNG and returns it.
+    #[wasm_bindgen(js_name = publicKey)]
+    pub fn public_key(&self) -> RequestPublicKey {
+        RequestPublicKey(self.0.public_key())
+    }
+
+    #[wasm_bindgen(js_name = diffieHellman)]
+    pub fn diffie_hellman(&self, their_public_key: &RequestPublicKey) -> RequestSharedSecret {
+        RequestSharedSecret(self.0.diffie_hellman(their_public_key.as_ref()))
+    }
+
+    #[allow(clippy::inherent_to_string)]
+    #[wasm_bindgen(js_name = toString)]
+    pub fn to_string(&self) -> String {
+        format!("{}", self.0)
+    }
```

Comment:
Why do we implement `to_string` here? Is it to display or to serialize these structs? If the former, maybe we could consider using [`#[wasm_bindgen(inspectable)]`](https://rustwasm.github.io/docs/wasm-bindgen/reference/attributes/on-rust-exports/inspectable.html#inspectable)

Reply:
I wasn't aware that it existed. But there is a slight difference here I think - `toString()` now returns the `Display` impl (that is, it is analogous to `str()` in Python), but `inspectable` would return something analogous to `repr()` in Python or `Debug` in Rust. Depends on what exactly you want from this method, I guess.